snapshot/src/main/java/io/zeebe/snapshots/impl/FileBasedSnapshotStore.java

```diff
@@ -87,10 +85,21 @@ public String getName() {
   }
 
   @Override
-  public void close() {
+  protected void onActorStarting() {
+    currentPersistedSnapshotRef.set(loadLatestSnapshot(snapshotsDirectory));
+    purgePendingSnapshotsDirectory();
+  }
+
+  @Override
+  protected void onActorClosing() {
     listeners.clear();
   }
 
+  @Override
+  public void close() {
+    closeAsync().join(10, TimeUnit.SECONDS);
```

Comment:
It implements PersistedSnapshotStore which extends CloseableSilently which has a close method. We could have it do nothing, but it can be useful to close from outside of the actor scheduler. We do this in a few other places, such as the log stream.

Reply:
I mean, `Actor::close` implements the same behavior, but with a different timeout. So that's why I was wondering, if it is necessary to override this.

Reply:
Ah, sorry, hadn't realized! Then I will remove it :+1: